osf/management/commands/migraterelations.py

```diff
@@ -437,79 +445,87 @@ def migrate_node_through_models(self):
+                                visible = clean_user_guid in modm_obj.visible_contributor_ids
+                                if (self.modm_to_django[clean_user_guid], self.modm_to_django[clean_node_guid]) not in contributor_hashes:
+                                    contributors.append(
+                                        Contributor(
+                                            read=read,
+                                            write=write,
+                                            admin=admin,
+                                            user_id=self.modm_to_django[clean_user_guid],
+                                            node_id=self.modm_to_django[clean_node_guid],
+                                            _order=order,
+                                            visible=visible
+                                        )
                                     )
-                                )
-                            node_relation_count += 1
-
-                        if count % page_size == 0 or count == total:
-                            NodeRelation.objects.bulk_create(node_relations)
-                            print('Through {} nodes and {} node relations, '
-                                  'saved {} NodeRelations'
-                                  .format(count, node_relation_count, len(node_relations)))
-                            contributors = []
-                            modm_obj._cache.clear()
-                            modm_obj._object_cache.clear()
-                            print('Took out {} trashes'.format(gc.collect()))
+                                    contributor_hashes.add((self.modm_to_django[clean_user_guid],
+                                                self.modm_to_django[clean_node_guid]))
+                                    order += 1
+                                    contributor_count += 1
+                                else:
+                                    logger.info('({},{}) already in contributor_hashes.'.format(
+                                        self.modm_to_django[clean_user_guid],
+                                        self.modm_to_django[clean_node_guid]))
+                            count += 1
+
+                            if count % page_size == 0 or count == total:
+                                Contributor.objects.bulk_create(contributors)
+                                logger.info('Through {} nodes and {} contributors, '
+                                      'saved {} contributors'.format(count, contributor_count, len(contributors)))
+                                contributors = []
+                                modm_obj._cache.clear()
+                                modm_obj._object_cache.clear()
+                                logger.info('Took out {} trashes'.format(gc.collect()))
+
+                            node_relations = []
```

Comment:
I think this should be declared in the outer scope with `contributors`.  Otherwise, only `node_relations` for the last page of nodes (if there are any) will be created.